vendor/vendor-android/adam/src/main/kotlin/com/malinskiy/marathon/android/adam/AdamAndroidDevice.kt

```diff
@@ -123,18 +124,23 @@ class AdamAndroidDevice(
             val local = File(localFilePath)
 
             measureFileTransfer(local) {
-                val channel = client.execute(
-                    CompatPullFileRequest(remoteFilePath, local, supportedFeatures, coroutineScope = this),
-                    serial = adbSerial
-                )
-                for (update in channel) {
-                    progress = update
+                val stat = client.execute(CompatStatFileRequest(remoteFilePath, supportedFeatures), serialNumber)
```

Comment:
Saves a lot of time so that we don't pull a file that doesn't exist

Reply:
How does this request actually work? What is the adb command used?

Reply:
There is no adb command to issue a stat request. It's an internal adb request